File: game-frontend/src/Model/game.ts
import {Tile} from './tile';
import {Block} from './tile';
import {Map} from './map';
import {Player} from './player';
import {Storage} from './storage';

export interface Game {

  updateScore(tile: Tile, block: Block): number;
  updateScoreLimit(): number;
  startGame(mapName: string): void;
  endGame(exit: boolean): boolean;
  computeScore(tile: Tile, block: Block): number;
  increaseScore(block: Block): number;
  getMaps(): Array<Map>;
  postNewMap(map: Map): void;
  getPlayer(): Player;
  setMap(map: string): Map;
  setNewMap(map: string): void;
  getStorage(): Storage;
  getCurrentMap(): Map;
  // playTurn(command: Command): void;
  getScore(): number;
  getScorLimit(): number;
  // UndoRedoCollector getCollector();
  setScore(score: number): void;
  setScoreLimit(limit: number): void;
  decreaseScoreLimit(): void;
  getMapsName(): Array<string>;
}

File: game-backend/src/main/java/game/model/Game.java
package game.model;

import game.Commands.Command;
import game.Commands.UndoRedoCollector;

import java.io.IOException;
import java.util.List;

public interface Game {
    int updateScore(Tile tile, Block block);
    int updateScoreLimit();
    void startGame(String mapName) throws IOException;
    boolean endGame(boolean exit);
    int computeScore(Tile tile, Block block);
    int increaseScore(final Block block);
    List<Map> getMaps() throws IOException;
    void postNewMap(Map map) throws IOException;
    Player getPlayer();
    Map setMap(String map);
    void setNewMap(String map) throws IOException;
    Storage getStorage();
    Map getCurrentMap();
    void playTurn(final Command command);
    int getScore();
    int getScorLimit();
    UndoRedoCollector getCollector();
    void setScore(int score);
    void decreaseScoreLimit();
}
